lib/ueberauth/strategy/amco/plugs/authenticated_user.ex

```diff
@@ -0,0 +1,56 @@
+defmodule Ueberauth.Strategy.Amco.Plugs.AuthenticatedUser do
+  alias Plug.Conn
+  alias Ueberauth.Strategy.Amco.API
+
+  @moduledoc """
+  This plug is in charge to verify and validate the access token
+  provided in the request against the OpenID Connect Provider.
+  It will try to get the access token from headers for json requests,
+  otherwise from cookies.
+
+  If the access token is valid, the current user will be assigned to the
+  connection and the request will continue as normal. In the other hand,
+  it will call the access_token_error function in the response handler.
+  """
+
+  def init(opts), do: opts
+
+  def call(%Conn{} = conn, opts) do
+    format = Keyword.get(opts, :format, :html)
+    handler = Keyword.fetch!(opts, :response_handler)
+
+    with {:ok, token} <- get_access_token(conn, format),
+         {:ok, response} <- validate_access_token(token) do
+
+      if Keyword.has_key?(handler.__info__(:functions), :access_token_success) do
+        handler.access_token_success(conn, response)
+      end
+
+      conn
```

Comment:
can the logic inside the with block be in a different function, I find confusing reading if end and then else, also the logic inside the if condition if kind of complex, you are retrieving the :functions key from the handler info, and then checking if it has a the access token success key, IMO that alone deserves its own function

Reply:
I removed the success callback because it is not required by the moment. Thanks.